channeldb/payment_control.go

```diff
@@ -229,7 +229,7 @@ func (p *PaymentControl) InitPayment(paymentHash lntypes.Hash,
 		return bucket.Delete(paymentFailInfoKey)
 	})
 	if err != nil {
-		return err
+		return fmt.Errorf("unable to init payment: %w", err)
```

Comment:
This doesn't seem to match the commit message anymore... Did we find out the cause for the wrapping to cause issues and changed it back?

Reply:
Yeah I think the other PR had some incremental changes that were not cleaned up so this commit doesn't make sense anymore. I'm squashing it with the "wrap all errors" one.

Reply:
IIRC we had to _revert_ this part to have the error introspection work properly: https://github.com/lightningnetwork/lnd/pull/8529#discussion_r1516997708